js/src/components/google-combo-account-card/claim-ads-account/claim-ads-account.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * External dependencies
+ */
+import { useCallback, useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import ClaimAccountButton from '.~/components/google-ads-account-card/claim-account-button';
+import Section from '.~/wcdl/section';
+import useGoogleAdsAccountStatus from '.~/hooks/useGoogleAdsAccountStatus';
+import { useAppDispatch } from '.~/data';
+import useGoogleAdsAccount from '.~/hooks/useGoogleAdsAccount';
+import useWindowFocusCallbackIntervalEffect from '.~/hooks/useWindowFocusCallbackIntervalEffect';
+import './claim-ads-account.scss';
+
+/**
+ * ClaimAdsAccount component.
+ *
+ * @return {JSX.Element} ClaimAdsAccount component.
+ */
+const ClaimAdsAccount = () => {
+	const [ updating, setUpdating ] = useState( false );
+	const { googleAdsAccount } = useGoogleAdsAccount();
+	const { hasAccess } = useGoogleAdsAccountStatus();
+	const { fetchGoogleAdsAccountStatus } = useAppDispatch();
+
+	const shouldClaimGoogleAdsAccount = Boolean(
+		googleAdsAccount?.id && hasAccess === false
+	);
+
+	const checkUpdatedAdsAccountStatus = useCallback( async () => {
+		if ( ! shouldClaimGoogleAdsAccount ) {
+			return;
+		}
+
+		await fetchGoogleAdsAccountStatus();
+	}, [ fetchGoogleAdsAccountStatus, shouldClaimGoogleAdsAccount ] );
+
+	useWindowFocusCallbackIntervalEffect( checkUpdatedAdsAccountStatus, 30 );
+
+	const handleOnClick = () => {
+		setUpdating( true );
+	};
+
+	return (
+		<Section.Card.Body className="gla-claim-ads-account-section">
+			<div className="gla-claim-ads-account-box">
+				<h4>
+					{ __(
+						'Claim your Google Ads account',
+						'google-listings-and-ads'
+					) }
+				</h4>
+				<p>
+					{ __(
+						'You need to accept the invitation to the Google Ads account we created for you. This gives you access to Google Ads and sets up conversion measurement. You must claim your account in the next 20 days.',
+						'google-listings-and-ads'
+					) }
+				</p>
+				<p className="gla-ads-post-claim-instructions">
+					{ __(
+						'After accepting the invitation, you’ll be prompted to set up billing. We highly recommend doing this to avoid having to do it later on.',
+						'google-listings-and-ads'
+					) }
+				</p>
+				<ClaimAccountButton
+					loading={ updating }
+					text={
+						updating
+							? __( 'Updating…', 'google-listings-and-ads' )
```

Comment:
I'm not sure "Updating..." is the best copy for this situation. In the current UI, we show "Updating..." instead of the "Claim..." button based on the loading prop returned from the `useUpsertAdsAccount()` hook. If we want to leave the button in a "pending" state while we're polling for updates to the account-status endpoint, we could use a word like "Waiting..." or "Checking access...". What do you think?

Reply:
Sure, I've updated the wording to "Waiting..."